### PR TITLE
Add InstanceNorm layer

### PR DESCRIPTION
**Pull request by Vimalakirti**

### CODE REVIEW

File: src/layer/norm.rs [hunk 1]
```diff
@@ -213,3 +215,271 @@ impl Layer for BatchNormLayer {
     (graph, vec![input_shapes[0].clone()])
   }
 }
+
+// InstanceNorm is a struct that represents an instance normalization layer, which computes
+// Y = (X - mean) * scale / sqrt(var + epsilon) + bias,
+// where mean and var are computed per instance per channel
+pub struct InstanceNormLayer;
+impl Layer for InstanceNormLayer {
```
Comment (liliatangxy) [inline src/layer/norm.rs:223]: Is there a lot of repeated code between InstanceNorm and BatchNorm?

Comment (Vimalakirti) [reply src/layer/norm.rs:223]: The main differences are the calculation for mean, variance, and the division per instance per channel (BatchNorm is per channel). But I agree there are some repeated codes such as addition of epsilon and multiplication with scale. Do you think it's better to introduce a macro for them?

Comment (liliatangxy) [reply src/layer/norm.rs:223]: No, I think it's different enough then

**Review by ddkang — APPROVED** [review]
lgtm if @liliatangxy approves

**Review by liliatangxy — APPROVED** [review]

**Pull request merged**